TodoApp.ConsoleApp/UI/UserInput.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+
+namespace TodoApp.ConsoleApp.UI
+{
+    public static class UserInput
+    {
+        const string Yes = "yes";
+        const string No = "no";
+
+        private static string ReadInput(bool required = false, string defaultValue = "")
+        {
+            string text = "";
+            if (required)
+            {
+                text = Console.ReadLine();
+
+                while (string.IsNullOrEmpty(text))
+                {
+                    Console.WriteLine("This field can't be empty!");
+                    text = Console.ReadLine();
+                }
+            }
+
+            else
+            {
+                text = Console.ReadLine();
+
+                if (string.IsNullOrEmpty(text))
+                {
+                    text = defaultValue;
+                }
+            }
+
+            return text;
+        }
+
+        public static string ReadText(string question, bool required = false, string defaultValue = "")
+        {
+            Console.WriteLine(question);
+            string text = "";
+            if (required)
+            {
+                text = ReadInput(required);
+
+                if (string.IsNullOrEmpty(text))
+                {
+                    text = defaultValue;
+                }
+            }
+
+            else
+            {
+                text = defaultValue;
+            }
+
+            return text;
+        }
+
+        public static int ReadOption(string question, string[] availableOptions, bool required = false, int defaultValue = 0)
+        {
+            Console.WriteLine(question);
+            int index = 1;
+            foreach (var option in availableOptions)
+            {
+                Console.WriteLine($"\t{index}: {option}");
+                index++;
+            }
+
+            index = 1;
```

Comment:
Create a different variable. It is bad practice to use the same variable for two different things.
I made the same mistake in the pseudo code :(

To fix this, you can use a regular `for` cycle with `i`, instead of `foreach`. And use `index` here.